saltproc/abc.py

```diff
@@ -99,21 +102,37 @@ def read_depleted_materials(self, read_at_end=False):
                 :class:`Materialflow` object holding material composition and properties.
 
         """
-
-    @abstractmethod
-    def run_depletion_step(self, mpi_args, threads):
+    def run_depletion_step(self, mpi_args, args):
         """Runs a depletion step as a subprocess with the given parameters.
 
         Parameters
         ----------
         mpi_args : list of str
             Arguments for running simulations on supercomputers using
             ``mpiexec`` or similar programs.
-        threads : int
+        args : int
             Threads to use for shared-memory parallelism
 
         """
 
+        print('Running %s' % (self.codename))
+        try:
+            if mpi_args is None:
+                stdout = sys.stdout
+            else:
+                stdout = None
+            subprocess.run(
+                args,
+                check=True,
+                cwd=self.output_path,
+                stdout=stdout,
+                stderr=subprocess.STDOUT)
```

Comment:
This may cause issues on windows machines, FYI. 
See related `watts` [issue](https://github.com/watts-dev/watts/issues/51).

Reply:
Could be okay if you're not using PIPE.